### ISSUE TITLE
[Suggestion] Mod Editor - Add common logic for Save Button

### ISSUE THREAD

**Issue by lmiol** [sedenion/OpenModMan#92]
Мany times i want to click Save button but it was inactive. Sometimes i thought there is a bug. But no. Aftet zoom interface on screen i figured out - Save button is inactive until you open a saved mod or do "Save as..." in new mod

99% of apps what we using every day "Save" button is active every time. And if you open a new document first time "Save" button works as "Save as.." and only after saving it works as usual as "Save". 
![image](https://github.com/sedenion/OpenModMan/assets/264879/a123d6c2-6799-47b7-9374-8bb0abb72497)

For reference try any text editor (notepad for example) or image editor. It will be hard to find inactive save button
![image](https://github.com/sedenion/OpenModMan/assets/264879/36c3c81f-c005-4c12-ba1d-c1f577fd41cc)

Regards

**Low priority**

**Comment by sedenion** [comment]
Changed "Save" and "Save as.." menu and toobar icon logic. "Save" menu and toolbar icon still disabled when no changes occurred since the last *save*, but is enabled by after "Build from directory", finally the "Save" toolbar button automatically perform "Save as.." if required.
